engines/bastion_katello/app/assets/javascripts/bastion_katello/products/details/repositories/details/repository-details-info.controller.js

```diff
@@ -14,194 +14,208 @@
                 });
-                deferred.resolve(results);
-            });
-            return deferred.promise;
-        };
+                return deferred.promise;
+            };
 
-        $scope.save = function (repository, saveUpstreamAuth) {
-            var deferred = $q.defer();
+            $scope.save = function (repository, saveUpstreamAuth) {
+                var deferred = $q.defer();
 
-            if (!saveUpstreamAuth) {
-                delete repository.upstream_username;
-                delete repository.upstream_password;
-            }
+                if (!saveUpstreamAuth) {
+                    delete repository.upstream_username;
+                    delete repository.upstream_password;
+                }
 
-            if (!_.isEmpty(repository.commaTagsWhitelist)) {
-                repository["docker_tags_whitelist"] = repository.commaTagsWhitelist.split(",").map(function(tag) {
-                    return tag.trim();
-                });
-            } else {
-                repository["docker_tags_whitelist"] = [];
-            }
-
-            repository.$update(function (response) {
-                deferred.resolve(response);
-                if (!_.isEmpty(response["docker_tags_whitelist"])) {
-                    repository.commaTagsWhitelist = repository["docker_tags_whitelist"].join(", ");
+                if (!_.isEmpty(repository.commaTagsWhitelist)) {
+                    repository["docker_tags_whitelist"] = repository.commaTagsWhitelist.split(",").map(function(tag) {
+                        return tag.trim();
+                    });
                 } else {
-                    repository.commaTagsWhitelist = null;
+                    repository["docker_tags_whitelist"] = [];
                 }
-                Notification.setSuccessMessage(translate('Repository Saved.'));
-            }, function (response) {
-                deferred.reject(response);
-                _.each(response.data.errors, function (errorMessage) {
-                    Notification.setErrorMessage(translate("An error occurred saving the Repository: ") + errorMessage);
+                /* eslint-disable camelcase */
+                repository.os_versions = $scope.osVersionsParam();
+                repository.$update(function (response) {
+                    deferred.resolve(response);
+                    if (!_.isEmpty(response["docker_tags_whitelist"])) {
+                        repository.commaTagsWhitelist = repository["docker_tags_whitelist"].join(", ");
+                    } else {
+                        repository.commaTagsWhitelist = null;
+                    }
+                    Notification.setSuccessMessage(translate('Repository Saved.'));
+                }, function (response) {
+                    deferred.reject(response);
+                    Notification.setErrorMessage(translate("An error occurred saving the Repository: ") + response.data.displayMessage);
```

Comment:
Was this not working as it was written before?

Reply:
It was displaying the notification as, for example:
```
An error occurred saving the Repository: invalid: Repositories can only require one OS version.
```
With this change it displays as
```
An error occurred saving the Repository: Os versions invalid: Repositories can only require one OS version.
```